### DISCUSSION TITLE
Scripter subroutine parameter undefined

### DISCUSSION THREAD

**Discussion by RedBlackThunder** [arendst/Tasmota#14957] [Q&A] [answered]
### PROBLEM DESCRIPTION
If using subroutines in scripter the parameters of the subroutine are undefined.

### REQUESTED INFORMATION
_Make sure your have performed every step and checked the applicable boxes before submitting your issue. Thank you!_

- [x] Read the [Contributing Guide and Policy](https://github.com/arendst/Tasmota/blob/development/CONTRIBUTING.md) and [the Code of Conduct](https://github.com/arendst/Tasmota/blob/development/CODE_OF_CONDUCT.md)
- [x] Searched the problem in [issues](https://github.com/arendst/Tasmota/issues)
- [x] Searched the problem in [discussions](https://github.com/arendst/Tasmota/discussions)
- [x] Searched the problem in the [docs](https://tasmota.github.io/docs/FAQ)
- [ ] Searched the problem in the [chat](https://discord.gg/Ks2Kzd4)
- [ ] Device used (e.g., Sonoff Basic): _____
- [x] Tasmota binary firmware version number used: 11.0.0 with scripter patch from 11.0.2
  - [ ] Pre-compiled
  - [x] Self-compiled
- [ ] Flashing tools used: _____
- [ ] Provide the output of command: `Backlog Template; Module; GPIO 255`:
- [ ] If using rules, provide the output of this command: `Backlog Rule1; Rule2; Rule3`:
- [ ] Provide the output of this command: `Status 0`:
- [ ] Set `weblog` to 4 and then, when you experience your issue, provide the output of the Console log:

### TO REPRODUCE
```
#AUTO(par)
print AUTO executed: %par%
```

```
12:50:45.568 CMD: AUTO OFF
12:50:45.571 AUTO executed: ???
12:50:45.578 MQT: tasmota/EG/KU/Rollershutter/stat/RESULT = {"Command":"Unknown"}
12:50:49.827 CMD: AUTO ON
12:50:49.831 AUTO executed: ???
12:50:49.837 MQT: tasmota/EG/KU/Rollershutter/stat/RESULT = {"Command":"Unknown"}
12:51:34.031 CMD: AUTO 0
12:51:34.035 AUTO executed: ???
12:51:34.041 MQT: tasmota/EG/KU/Rollershutter/stat/RESULT = {"Command":"Unknown"}
12:51:37.484 CMD: AUTO 1
12:51:37.488 AUTO executed: ???
```

### EXPECTED BEHAVIOUR
Variable par set to parameter of the subrouting

**(Please, remember to close the issue when the problem has been addressed)**

**Comment by gemu2015** [comment, marked as answer]
There are no local variables in scripter. Thus also function Parameters have to be defined in >D

**Comment by RedBlackThunder** [comment]
My fault it is also mentioned in the documentation. I have no idea, why I haven't read it until now.
